src/jit/emitxarch.cpp

```diff
@@ -2315,6 +2315,8 @@ void emitter::emitIns(instruction ins)
     id->idInsFmt(fmt);
     id->idCodeSize(sz);
 
+    emitAttr attr = id->idOpSize();
+    CheckAndSetContainsAVX(ins, attr);
     dispIns(id);
```

Comment:
Adding this code in all these methods seems unnecessary. I believe there are methods that are always called to encode an AVX instruction, and we should attempt to localize this checking there.